CopyKeys/CopyKeys.ps1

```diff
@@ -1,1468 +1,1676 @@
+    $LoadingLabel.text = ""
+    $LoadingLabel.AutoSize = $true
+    $LoadingLabel.Enabled = $true
+    $LoadingLabel.Add_MouseHover({Show-Help})
+
+    $SelectButtonFormatting = @{"location"=@(184, 580); "width"=75; "height"=30}
+    Add-CommonFormatting -UiObject $SelectButton -Formattings $SelectButtonFormatting
+    $SelectButton.BackColor = "#eeeeee"
+    $SelectButton.text = "Select"
+    $SelectButton.Add_Click({Get-AllSelections})
+
+    $MsLogo = New-Object System.Windows.Forms.PictureBox
+    $MsLogo.width = 140
+    $MsLogo.height = 80
+    $MsLogo.location = New-Object System.Drawing.Point(150, 10)
+    $MsLogo.imageLocation = "https://c.s-microsoft.com/en-us/CMSImages/ImgOne.jpg?version=D418E733-821C-244F-37F9-DC865BDEFEC0"
+    $MsLogo.SizeMode = [System.Windows.Forms.PictureBoxSizeMode]::zoom
+
+    # Populating the subscription dropdown and launching the form
+
+    [array]$SubscriptionArray = ((Get-AzSubscription).Name | Sort-Object)
+
+    foreach ($Item in $SubscriptionArray)
+    {
+        $SuppressOutput = $SubscriptionDropDown.Items.Add($Item)
+    }
+
+    $Longest = ($SubscriptionArray | Sort-Object Length -Descending)[0]
+    $SubscriptionDropDown.DropDownWidth = ([System.Windows.Forms.TextRenderer]::MeasureText($Longest, `
+        $SubscriptionDropDown.Font).Width, $SubscriptionDropDown.Width | Measure-Object -Maximum).Maximum
+
+    $UserInputForm.controls.AddRange($FormElements.Values + $LoadingLabel)
+    $UserInputForm.controls.AddRange($MsLogo)
+    [void]$UserInputForm.ShowDialog()
+}
+#endregion
+
+### <summary>
+### Encrypts the secret based on the key provided.
+### </summary>
+### <param name="DecryptedValue">Decrypted secret value.</param>
+### <param name="EncryptedAlgorithm">Name of the encryption algorithm used.</param>
+### <param name="AccessToken">Access token for the key vault.</param>
+### <param name="KeyId">Id of the key to be used for encryption.</param>
+function Encrypt-Secret(
+    $DecryptedValue,
+    [string]$EncryptedAlgorithm,
+    [string]$AccessToken,
+    [string]$KeyId)
+{
+    $Body = @{
+        'value' = $DecryptedValue
+        'alg'   = $EncryptedAlgorithm}
+
+    $BodyJson = ConvertTo-Json -InputObject $Body
+
+    $Params = @{
+        ContentType = [ConstantStrings]::contentType
+        Headers     = @{
+            [ConstantStrings]::authHeader = [ConstantStrings]::tokenType + " " + $AccessToken }
+        Method      = [ConstantStrings]::httpPost
+        URI         = "$KeyId" + '/encrypt?api-version=7.1'
+        Body        = $BodyJson}
+
+    try
+    {
+        $OutputLogger.Log(
+            $MyInvocation,
+            "Starting REST call - " + $Params.URI,
+            [LogType]::INFO)
+        $Response = Invoke-RestMethod @Params
+    }
+    catch
+    {
+        $errorStr = Out-String -InputObject $PSItem
+
+        Write-Verbose "`nEncrypt failure: `n$errorStr"
+        throw [Errors]::SecretEncryptionFailed($errorStr)
```

Comment:
will this error message be shown to user only -Verbose option....or by default?

Reply:
and have we tested that proper error message is being shown , if secret decryption fails?

Reply:
No need for -Verbose. The error will be shown always and will get logged in the file as well.
Yes, we've tested this and were able to identify an issue as well.